### DISCUSSION TITLE
Why Fooocus SD 1.5 refiner result in weird pattern artifact swirl thing?

### DISCUSSION THREAD

**Discussion by orangpelupa** [lllyasviel/Fooocus#2290] [Q&A]
**Read Troubleshoot**

[x] I confirm that I have read the [Troubleshoot](https://github.com/lllyasviel/Fooocus/blob/main/troubleshoot.md) guide before making this issue.

**Describe the problem**
So ive been using fooocus and ever since i used SD 1.5 refiner, some seeds will result in weird patterns. While some others are clean.
![image](https://github.com/lllyasviel/Fooocus/assets/16116746/d7f60dd3-3867-487b-a168-b685d1f3b706)

with SD1.5 refiner
![image](https://github.com/lllyasviel/Fooocus/assets/16116746/7e3ee322-746e-45b3-a7f3-2b39f2a6c791)

withuot refginer
![image](https://github.com/lllyasviel/Fooocus/assets/16116746/a6ca96a2-6be0-476b-9f74-273c15313724)

i've tested the same seed and checkpoints and prompt with other GUI and didnt result in that weird artifact pattern swirl thing

the settings 
{
"Prompt": "photo of a happy beautiful girl sitting in on a bed, in a bedroom with white wall, wearing a white tshirt and blue jeans",
"Negative Prompt": "unrealistic, saturated, high contrast, big nose, painting, drawing, sketch, cartoon, anime, manga, render, CG, 3d, watermark, signature, label, porn, naked, nude, topless, skimpy, deformed, midget, dwarf, three arms, three hands, weird arms, ugly arms, deformed hands, lifeless, expressionless, three legs, ugly legs, deformed legs, exposes breast, exposed breasts, exposed breast, exposed nipples, deformed ankle",
"Fooocus V2 Expansion": "photo of a happy beautiful girl sitting in on a bed, in a bedroom with white wall, wearing a white tshirt and blue jeans, highly detailed, cinematic, extremely lush, colorful, light, sharp focus, very intricate, elegant, dynamic colors, ambient, radiant, professional, winning, color, creative, positive, vibrant, attractive, confident, emotional, majestic",
"Styles": "['Fooocus V2', 'Fooocus Negative', 'SAI Photographic']",
"Performance": "Quality",
"Resolution": "(896, 1152)",
"Sharpness": 2,
"Guidance Scale": 3,
"ADM Guidance": "(1.5, 0.8, 0.3)",
"Base Model": "leosamsHelloworldXL_helloworldXL50GPT4V.safetensors",
"Refiner Model": "majicmixLux_v3.safetensors",
"Refiner Switch": 0.5,
"Sampler": "dpmpp_2m_sde_gpu",
"Scheduler": "karras",
"Seed": 6721681171047527111,
"Version": "v2.1.864"
}

**Full Console Log**
[System ARGV] ['B:\\tools\\StabilityMatrix-win-x64\\Data\\Packages\\Fooocus\\launch.py', '--preset', 'realistic']
Python 3.10.11 (tags/v3.10.11:7d4cc5a, Apr  5 2023, 00:38:17) [MSC v.1929 64 bit (AMD64)]
Fooocus version: 2.1.864
Loaded preset: B:\tools\StabilityMatrix-win-x64\Data\Packages\Fooocus\presets\realistic.json
Running on local URL:  http://127.0.0.1:7865

To create a public link, set `share=True` in `launch()`.
Total VRAM 8192 MB, total RAM 32694 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 3070 : native
VAE dtype: torch.bfloat16
Using pytorch cross attention
Refiner unloaded.
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale'}
Base model loaded: B:\tools\StabilityMatrix-win-x64\Data\Models\StableDiffusion\realisticStockPhoto_v20.safetensors
Request to load LoRAs [['SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors', 0.25], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [B:\tools\StabilityMatrix-win-x64\Data\Models\StableDiffusion\realisticStockPhoto_v20.safetensors].
Loaded LoRA [B:\tools\StabilityMatrix-win-x64\Data\Packages\Fooocus\models\loras\SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors] for UNet [B:\tools\StabilityMatrix-win-x64\Data\Models\StableDiffusion\realisticStockPhoto_v20.safetensors] with 788 keys at weight 0.25.
Loaded LoRA [B:\tools\StabilityMatrix-win-x64\Data\Packages\Fooocus\models\loras\SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors] for CLIP [B:\tools\StabilityMatrix-win-x64\Data\Models\StableDiffusion\realisticStockPhoto_v20.safetensors] with 264 keys at weight 0.25.
Fooocus V2 Expansion: Vocab with 642 words.
Fooocus Expansion engine loaded for cuda:0, use_fp16 = True.
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
[Fooocus Model Management] Moving model(s) has taken 1.70 seconds
App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 3.0
[Parameters] Seed = 6721681171047527111
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 60 - 30
[Fooocus] Initializing ...
[Fooocus] Loading models ...
model_type EPS
UNet ADM Dimension 0
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_l.logit_scale'}
left over keys: dict_keys(['alphas_cumprod', 'alphas_cumprod_prev', 'betas', 'log_one_minus_alphas_cumprod', 'posterior_log_variance_clipped', 'posterior_mean_coef1', 'posterior_mean_coef2', 'posterior_variance', 'sqrt_alphas_cumprod', 'sqrt_one_minus_alphas_cumprod', 'sqrt_recip_alphas_cumprod', 'sqrt_recipm1_alphas_cumprod'])
Refiner model loaded: B:\tools\StabilityMatrix-win-x64\Data\Models\StableDiffusion\majicmixLux_v3.safetensors
model_type EPS
UNet ADM Dimension 2816
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
extra {'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.logit_scale'}
Base model loaded: B:\tools\StabilityMatrix-win-x64\Data\Models\StableDiffusion\leosamsHelloworldXL_helloworldXL50GPT4V.safetensors
Request to load LoRAs [['None', 0.25], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [B:\tools\StabilityMatrix-win-x64\Data\Models\StableDiffusion\leosamsHelloworldXL_helloworldXL50GPT4V.safetensors].
Request to load LoRAs [['None', 0.25], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [B:\tools\StabilityMatrix-win-x64\Data\Models\StableDiffusion\majicmixLux_v3.safetensors].
Requested to load SDXLClipModel
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.68 seconds
[Fooocus] Processing prompts ...
[Fooocus] Preparing Fooocus text #1 ...
[Prompt Expansion] photo of a happy beautiful girl sitting in on a bed, in a bedroom with white wall, wearing a white tshirt and blue jeans, highly detailed, cinematic, extremely lush, colorful, light, sharp focus, very intricate, elegant, dynamic colors, ambient, radiant, professional, winning, color, creative, positive, vibrant, attractive, confident, emotional, majestic
[Fooocus] Encoding positive #1 ...
[Fooocus] Encoding negative #1 ...
[Parameters] Denoising Strength = 1.0
[Parameters] Initial Latent shape: Image Space (1152, 896)
Preparation time: 56.07 seconds
[Sampler] refiner_swap_method = vae
[Sampler] sigma_min = 0.0291671771556139, sigma_max = 14.614643096923828
Requested to load SDXL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 1.32 seconds
100%|██████████| 30/30 [00:15<00:00,  1.89it/s]
Fooocus VAE-based swap.
Requested to load Interposer
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.95 seconds
Requested to load BaseModel
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.25 seconds
100%|██████████| 30/30 [00:13<00:00,  2.16it/s]
Requested to load AutoencoderKL
Loading 1 new model
[Fooocus Model Management] Moving model(s) has taken 0.61 seconds
Image generated with private log at: B:\tools\StabilityMatrix-win-x64\Data\Packages\Fooocus\outputs\2024-02-12\log.html
Generating and saving time: 34.36 seconds
Requested to load SDXLClipModel
Requested to load GPT2LMHeadModel
Loading 2 new models
[Fooocus Model Management] Moving model(s) has taken 0.38 seconds
Total time: 91.28 seconds

**Comment by mashb1t** [comment]
This might have to do with the training of the refiner model, maybe it doesn't include high res photos or you're trying to create resolutions outside its training region, which leads to these artifacts.
Not really anything Fooocus can fix.

**Comment by orangpelupa** [comment]
> This might have to do with the training of the refiner model, maybe it doesn't include high res photos or you're trying to create resolutions outside its training region, which leads to these artifacts. Not really anything Fooocus can fix.

the weird thing is that i've tested the same seed and checkpoints and prompt and settings (including resolution) with other GUI and didnt result in that weird artifact pattern swirl thing. it only appear in fooocus

**Comment by AdamM2024** [comment]
how do i use fooocus on my laptop i use cpu?

**Comment by mashb1t** [comment]
@AdamM2024 this is off-topic, but here is the answer anyways: use `--always-cpu`, depending on your OS in run.bat or entrypoint, directly after entry_with_update.py
Please open a discussion in the Q&A section next time.

**Comment by mashb1t** [comment]
@orangpelupa moved to discussion, might move to issues again when confirmed/not solved by other users
